airbyte-integrations/connectors/source-file/source_file/client.py

```diff
@@ -286,11 +286,15 @@ def load_dataframes(self, fp, skip_data=False) -> List:
             raise ConfigurationError(error_msg) from err
 
         reader_options = {**self._reader_options}
-        if skip_data and self._reader_format == "csv":
-            reader_options["nrows"] = 0
-            reader_options["index_col"] = 0
+        if self._reader_format == "csv":
+            reader_options["chunksize"] = 10000
+            if skip_data:
+                reader_options["nrows"] = 0
+                reader_options["index_col"] = 0
 
-        return [reader(fp, **reader_options)]
+            yield from reader(fp, **reader_options)
+        else:
+            yield reader(fp, **reader_options)
```

Comment:
you need to update return annotations -> `Iterable`

Reply:
Done.